packages/connect-popup/src/view/react/support/ReactWrapper.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { useEffect } from 'react';
+import styled, { ThemeProvider, StyleSheetManager } from 'styled-components';
+import { THEME } from '@trezor/components';
+
+const View = styled.div`
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
+    text-align: center;
+    min-height: 100vh;
+`;
+
+// todo: does 'support' folder make sense for this?
+// todo: how do we type react components in july 2022?
+interface Props {
+    children: React.ReactChildren;
+}
+
+export const ReactWrapper = (props: Props) => {
+    const styleSlot = document.getElementById('react');
+
+    if (!styleSlot?.shadowRoot) {
+        console.error('could not find shadow-root to mount react application');
+    }
+
+    useEffect(() => {
+        console.log('mount react wrapper');
+        return () => {
+            // document.getElementById('react')?.remove();
+            console.log('unmount react wrapper');
+        };
+    });
+
+    return (
+        // @ts-ignore. typings don't like using shadowRoot as target but it works
```

Comment:
use `@ts-expect-error` instead of ignore, it's better because if error will get fixed in future, it will throw that you can remove it

Reply:
good point 29f89c98e4d666089677f07cc5a6f3ae878563f5